django/contrib/humanize/templatetags/humanize.py

```diff
@@ -156,7 +158,7 @@ def _check_for_i18n(value, float_formatted, string_formatted):
 
     for exponent, converters in intword_converters:
         large_number = 10 ** exponent
-        if value < large_number * 1000:
+        if absval < large_number * 1000:
             new_value = value / large_number
             rounded_value = round_away_from_one(new_value)
             return _check_for_i18n(new_value, *converters(rounded_value))
```

Comment:
Some new tests which are added by you failed, for example:
`humanize_tests.tests.HumanizeTests.test_i18n_intword [-1000000] `

It can be fixed be:
```suggestion
            return _check_for_i18n(new_value, *converters(abs(rounded_value)))
```